pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -1432,3 +1441,14 @@ func (cb *ClusterBuilder) buildExternalSDSCluster(addr string) *cluster.Cluster
 	}
 	return c
 }
+
+// configureALPNOverride determines whether alpn_override should be added to metadata
+func configureALPNOverride(tlsMode networking.ClientTLSSettings_TLSmode, md *core.Metadata) *core.Metadata {
```

Comment:
I think you added a comment somewhere on the issue, but do you even need to return anything here since you're passing a pointer to md? In fact `AddALPNOverrideToMetadata` probably doesn't need a return value either if you're manipulating a pointer value

Reply:
Yes you do because if the metadata is nil, we want to return a new metadata with the headers. That's the only reason. Empty metadata can be modified and set, but not nil.

Reply:
Then why pass a pointer in the first place?

Reply:
Isn't that less memory on the stack?

Reply:
That's how it's stored in the mutable cluster.